common/workerpool.c

```diff
@@ -116,9 +116,8 @@ workerpool_t *workerpool_create(int nthreads)
         for (int i = 0; i < nthreads; i++) {
             int res = pthread_create(&wp->threads[i], NULL, worker_thread, wp);
             if (res != 0) {
-                debug_print("Insufficient system resources to create workerpool threads\n");
-                // errno already set to EAGAIN by pthread_create() failure
-                return NULL;
+                perror("pthread_create");
+                exit(-1);
```

Comment:
Don't print in the library in release mode. We added `debug_print` so that release builds will prevent stdout and stderr output. You also change the behaviour here by quitting the process instead of returning NULL. This was added in https://github.com/AprilRobotics/apriltag/pull/224 by design.